Remove old docs and update mkdocs and md files to reflect the new ver… (#83)
Harvard-Neutrino/prometheus: docs/installation/containers.md

# Install with Containers

> If you need to run simulations on a computing cluster, or the built-in installer doesn't work for you, pre-built Docker and Apptainer images with all dependencies are available on the [GitHub Container Registry](https://github.com/Harvard-Neutrino/prometheus/pkgs/container/prometheus). If you're on a personal machine, go with Docker; if you're on a cluster, Apptainer is usually the better fit (and is often already installed).

Reviewer: ```suggestion
If you need to run simulations on a computing cluster, or the built-in installer doesn't work for you, pre-built Docker and Apptainer images with all dependencies are available on the [GitHub Container Registry](https://github.com/Harvard-Neutrino/prometheus/pkgs/container/prometheus).

If you're on a personal machine, go with Docker; if you're on a cluster, Apptainer is usually the better fit (and is often already installed).
```

!!! note

>     The provided images are built for x86_64 and are not compatible with ARM-based architectures (e.g., Apple M-series Macs). If you are a Mac user on Apple Silicon, install from source instead.

Reviewer: With changes in this PR, we won't have the install from source section in the docs anymore. And the pre-built installer doesn't work for macs.

Should we make an install from source section for M-series mac users? (Which realistically will be most of Mac users these days). Or there is no point spending time accommodating this user base cause most of the time Prometheus is run on a cluster anyway?

In the latter case, we can include the short sentence about it instead, something like:

> The provided images are built for x86_64 and are not compatible with ARM-based architectures (e.g., Apple M-series Macs). Given that most people run Prometheus on a cluster, we chose not to prioritize Apple Silicon support at this time. If you would like it implemented, let us know by [opening an issue on GitHub](https://github.com/Harvard-Neutrino/prometheus/issues/new/choose).

Author: The install from source option will take a little bit (working on it), since it needs a complete rewrite of the installation script and methodology.

For now containers are the only way.

I'd leave as is and once I have an installer working for macs we can update.

Author: macOS now supported. Updated docs in recent commit

Author: This should say use the built in installer instead

Reviewer: Done

## Available Images

| Tag           | Description       |
| ------------- | ----------------- |
| `:latest`     | CPU-only image    |
| `:latest-gpu` | GPU-enabled image |
| `:VERSION`    | Specific release  |

## Using Docker

> If you need help getting started with Docker, see the [Docker documentation](https://docs.docker.com/desktop/). Here is an outline of the steps to set things up:

Reviewer: Two different thoughts.

```suggestion
If you need help getting started with Docker, see the [Docker documentation](https://docs.docker.com/desktop/). 

Here is an outline of the steps to set things up:
```

1. **Pull the image**

    ```sh
    docker pull ghcr.io/harvard-neutrino/prometheus:latest
    ```

2. **Run the container and launch a shell**

    ```sh
    docker run --rm -it ghcr.io/harvard-neutrino/prometheus:latest
    ```

3. **Run an example**

    ```sh
    docker run --rm -v "$PWD/output:/output" ghcr.io/harvard-neutrino/prometheus:latest python /opt/prometheus/examples/01_basic_water.py
    ```

>     Use volume mounting (`-v "$PWD/output:/output"`) to access output files on your local machine. For more container running options, refer to the [Docker containers documentation](https://docs.docker.com/engine/containers/run/).

Reviewer: Is this volume mounting note necessary? It's already a part of the command, so if the user is copying the command, they are already using it.

If it's an FYI on what the command does, we can say something like this. And also would be nice to very briefly explain why the user needs to know this. Not including this into the suggestion because I don't know why lol

```suggestion
    The `-v "$PWD/output:/output"` part of the command enables [volume mounting](https://docs.docker.com/engine/storage/volumes/#mounting-a-volume-over-existing-data) to access output files on your local machine.
    
For more container running options, refer to the [Docker containers documentation](https://docs.docker.com/engine/containers/run/).
```

Author: Definitely will be. Also the statement: "to access output files on your local machine" explains why they might want to do this.

### GPU Support

For GPU-accelerated simulations, you will need the [NVIDIA Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html). Once installed, use the GPU image:

```sh
docker run --rm -it --gpus all ghcr.io/harvard-neutrino/prometheus:latest-gpu
```

If this fails, check that your NVIDIA drivers and container toolkit are correctly installed.

## Using Apptainer / Singularity

Apptainer (formerly Singularity) is particularly useful for running simulations on computing clusters, where it is often pre-installed. If you need help getting started, see the [Apptainer documentation](https://apptainer.org/docs/).

1. **Load the Apptainer module** (if on a cluster)

    ```sh
    module load apptainer
    ```

2. **Pull the container image**

    ```sh

>     singularity pull docker://ghcr.io/harvard-neutrino/prometheus:latest

Reviewer: shouldn't the command be `apptainer pull ...` ? Looking at Apptainer Getting Started docs and seeing that all commands are `apptainer <do a thing>`

Author: Fixed in new commit. Note that the singularity pull also works. Just changed it for consistency
